packages/core/src/llm/router/router.ts

```diff
@@ -0,0 +1,432 @@
+import {
+  generateText,
+  LanguageModel,
+  Tool,
+  generateObject,
+  jsonSchema,
+  Schema,
+  CoreMessage,
+  streamText,
+  streamObject,
+} from 'ai';
+import { JSONSchema7 } from 'json-schema';
+import { z, ZodSchema } from 'zod';
+
+import { MastraPrimitives } from '../../action';
+import { ToolsInput } from '../../agent/types';
+import { delay } from '../../utils';
+import { MastraLLM } from '../base';
+import {
+  GenerateReturn,
+  LLMTextOptions,
+  LLMInnerStreamOptions,
+  LLMStreamObjectOptions,
+  LLMStreamOptions,
+  LLMTextObjectOptions,
+  StreamReturn,
+} from '../types';
+
+export class ModelRouter extends MastraLLM {
```

Comment:
ModelRouter, isn't that a weird name? Why call it router, is it because we're routing ai models?